---
title: "Log of 'willer_tc_prad_match' files" 
subtitle: "Project: thesis_tc_pgs_in_AD"
author: "Nathalie Nilsson"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r,echo=FALSE}
path.to.project <- paste(rprojroot::find_rstudio_root_file(), "/", sep = "")
path.to.wd <- gsub(rprojroot::find_rstudio_root_file(), "", getwd())
```
**Location of project directory:** `r path.to.project`

**Working directory relative to project:** `r path.to.wd`

## Input
Cleaned meta-analysis summary data:
```{r}
base.file <- "willer_tc_common_snps.txt"
base.file.woAPOE <- "willer_tc_common_snps_woAPOE.txt"
```

PREVENT-AD genetic data:
```{r}
prad.file <- "D:/OneDrive - McGill University/Lab_stuff/Research/Cohort_data/ADPREVENT/Genetic_data/PRAD_2019/imputed/prad_2019_imp_eur.bim"
```

## Libraries
```{r}
library(data.table)
```

## Read data
**Willer base data**
```{r}
# Read base data
base <- fread(base.file, header = TRUE)[, -c("CHR", "BP"), with = FALSE]
base
```

```{r}
# Base data without APOE
base.woAPOE <- fread(base.file.woAPOE, header = TRUE)[, -c("CHR", "BP"), with = FALSE]
base.woAPOE
```

**PREVENT-AD target data**
```{r}
prad <- fread(prad.file, header = FALSE, 
              col.names = c("CHR", "SNP", "CM", "BP", "REF.prad", "ALT.prad"))
prad[, POS := list(paste(CHR, BP, sep = ":"))]
prad
```

### SNP discrepancies
```{r}
# Merge data
m <- merge(base, prad, by = "POS", suffixes = c(".base", ".prad"))[
    ][, .(CHR, BP, SNP.base, SNP.prad, REF, REF.prad, ALT, ALT.prad, BETA, P)]

m.woAPOE <- merge(base.woAPOE, prad, by = "POS", suffixes = c(".base", ".prad"))[
    ][, .(CHR, BP, SNP.base, SNP.prad, REF, REF.prad, ALT, ALT.prad, BETA, P)]

# Calculate number of alleles for each variant
m$N.alleles <- apply(m[, .(REF,REF.prad,ALT,ALT.prad)], 1, function(x) length(unique(x)))
m

unique(m[, N.alleles])

# Identify SNPs to be flipped
correct <- m[REF == REF.prad & ALT == ALT.prad]
correct

a1a2mix <- m[REF == ALT.prad & ALT == REF.prad]
a1a2mix

flipped <- m[N.alleles == 4, ]
flipped

incorrect <- m[N.alleles == 3, 
               ][, EXCL_REASON := "Multiple alleles"]
incorrect

nrow(correct) + nrow(a1a2mix) + nrow(flipped) + nrow(incorrect) == nrow(m)
```

## Write data
```{r}
dir.create("willer_tc_prad_match")
fwrite(m[!SNP.prad %in% incorrect[, SNP.prad] , .(SNP = SNP.prad, REF, P, BETA, ALT)], 
       file = "willer_tc_prad_match/willer_tc_prad_match.txt", sep = "\t")
fwrite(m[!SNP.prad %in% incorrect[, SNP.prad], .(SNP = SNP.prad, P)], 
       file = "willer_tc_prad_match/willer_tc_prad_match_snps_clump.txt", sep = "\t")
fwrite(m.woAPOE[!SNP.prad %in% incorrect[, SNP.prad] , .(SNP = SNP.prad, REF, P, BETA, ALT)], 
       file = "willer_tc_prad_match/willer_tc_prad_match_woAPOE.txt", sep = "\t")
fwrite(flipped[, .(SNP.prad)], 
       file = "willer_tc_prad_match/willer_tc_prad_match_SNPsFlip.txt", 
       sep = "\t", col.names = FALSE)
fwrite(incorrect, 
       file = "willer_tc_prad_match/willer_tc_prad_match_SNPsExcl.txt", 
       sep = "\t", col.names = FALSE)
```