---
title: "Willer data plotted"
subtitle: "Project: thesis_tc_pgs_in_AD"
author: "Nathalie Nilsson"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document: 
    df_print: kable
    fig_height: 4
    fig_width: 6.5
    theme: paper
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r,echo=FALSE}
path.to.project <- paste(rprojroot::find_rstudio_root_file(), "/", sep = "")
path.to.wd <- gsub(rprojroot::find_rstudio_root_file(), "", getwd())
```
**Location of project directory:** `r path.to.project`

**Working directory relative to project:** `r path.to.wd`

## Input
```{r}
raw.file <- "raw_data/jointGwasMc_TC.txt"

common.file <- "willer_tc_common_snps.txt"
common.file.woapoe <- "willer_tc_common_snps_woAPOE.txt"

# relative to project directory:
prad.file <- "analyses/preventad/1_pg_scoring/target_files/prad_tc_pgs_LDPRUNED.bim"
```

## Libraries
```{r}
library(data.table)
library(ggplot2)

source(paste(path.to.project, "scripts/project_functions.R", sep = ""))
```

## Data

### PRAD SNPs after clumping
```{r}
prad.snps <- fread(paste(path.to.project, prad.file, sep = ""), 
                   header = F, col.names = c("CHR", "rsid", "CM", "BP", "A1", "A2"))[, POS := paste(CHR, BP, sep = ":")]
head(prad.snps)
```

### Raw willer data
```{r}
raw <- fread(raw.file)
head(raw)
```

### Common SNPs
```{r}
common <- fread(common.file)[, pos.n := as.numeric(gsub(":", ".", POS))]
head(common)

ggplot(common, aes(pos.n, -log10(P), color = factor(CHR))) +
    geom_point() +
    scale_color_viridis_d() +
    scale_x_continuous(breaks = 1:22) +
    my.theme +
    theme(legend.position = "none")
    
```

### Excluded SNPs
```{r}
excl <- raw[, POS := gsub("chr", "", SNP_hg19)
            ][, pos.n := as.numeric(gsub(":", ".", POS))
              ][, c("rsid", "P-value", "POS", "pos.n"), with = FALSE
                ][, c("CHR", "BP") := tstrsplit(POS, ":", fixed= TRUE, type.convert = TRUE)
                  ][!POS %in% common[, POS]]
colnames(excl) <- c("rsid", "P", "POS", "pos.n", "CHR", "BP")
head(excl)
excl[, .N]

ggplot(excl, aes(pos.n, -log10(P), color = factor(CHR))) +
    geom_point() +
    scale_color_viridis_d() +
    scale_x_continuous(breaks = 1:22) +
    my.theme +
    theme(legend.position = "none")
```

### SNPs left after clumping
```{r}
common.f <- common[POS %in% prad.snps[, POS]]

ggplot(common.f, aes(pos.n, -log10(P), color = factor(CHR))) +
    geom_point() +
    scale_color_viridis_d() +
    scale_x_continuous(breaks = 1:22) +
    my.theme +
    theme(legend.position = "none")
```

### SNPs remaining after clumping; APOE gene (+- 1Mbp) excluded
```{r}
common.woapoe <- fread(common.file.woapoe)[POS %in% prad.snps[, POS]
                                         ][, pos.n := as.numeric(gsub(":", ".", POS))]
head(common.woapoe)

ggplot(common.woapoe, aes(pos.n, -log10(P), color = factor(CHR))) +
    geom_point() +
    scale_color_viridis_d() +
    scale_x_continuous(breaks = 1:22) +
    my.theme +
    theme(legend.position = "none")
```

Top chromosome 19 SNP
```{r}
head(common.woapoe[CHR == 19, 
         ][order(P)])
```

